File: 1253.py
alfabeto = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
for i in range(0, int(input())):
    final = ''
    frase = input()
    n = int(input())
    for j in range(0, len(frase)):
        if alfabeto.index(frase[j]) - n < 0:
            final += alfabeto[alfabeto.index(frase[j]) - n + 26]
        else:
            final += alfabeto[alfabeto.index(frase[j]) - n]
    print(final)

File: 1253.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');
let alfabeto = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z'];
let final = ''; let frase = ''; let n = 0; let g = parseInt(lines.shift());
for(var i = 0; i < g; i++){
    final = '';
    frase = lines.shift();
    n = parseInt(lines.shift());
    for(var j = 0; j < frase.length; j++){
        if(alfabeto.indexOf(frase[j]) - n < 0){
            final += alfabeto[alfabeto.indexOf(frase[j]) - n + 26];
        }
        else{
            final += alfabeto[alfabeto.indexOf(frase[j]) - n];
        }
    }
    console.log(final);
}
